homeassistant/helpers/template.py

```diff
@@ -1964,6 +1982,29 @@ def __bool__(self):
         return super().__bool__()
 
 
+@dataclass
+class Ext:
+    """Jinja extension definition."""
```

Comment:
I'm not sure I see the advantage of creating an array of Ext objects as opposed to just a set of helper functions for assigning stuff.

Is there something odd about providing optional parameters for a func that's different with the obj?  If that's the case then I see the advantage of this method.  Otherwise I think the helper functions pattern would be cleaner and more understandable.

Reply:
That's just where my brain went first, but I can see your point.  I reworked the whole thing to simply call a helper function that it's both working and arguably clearer.  I'll upload the new changes tomorrow.

Incidentally, as you might imagine, the issue that this PR is intended to address only requires minor changes to the `device_attr` function that are totally unrelated to restructuring the way extensions are added.  Everything that you reviewed is really part of https://github.com/home-assistant/core/pull/76006, which fixes a larger issue preventing many tests & filters from being used as intended.  My intention was to get that PR merged, and then this would build on that.  To that end, I'll upload a new commit based on your comments in https://github.com/home-assistant/core/pull/76006.

Thanks for reviewing!